tests/agent_tests/agent_tests.cc

```diff
@@ -78,3 +79,109 @@ TEST_P(AgentTests, SetThenGetEconParam) {
+  double finance_eps = 1e-4;
+
+  // testing by "round trip" with already tested functions
+  EXPECT_NEAR(P, agent_->PV(n,0,agent_->FV(n,0,P,0),0), finance_eps);  // discount rate = 0
+  EXPECT_NEAR(P, agent_->PV(0,i,agent_->FV(0,i,P,0),0), finance_eps);  // discount time = 0
+  EXPECT_NEAR(P, agent_->PV(i,i,agent_->FV(1,i,P,0),0), finance_eps);  // discount time = 1
+  EXPECT_NEAR(P, agent_->PV(n,i,agent_->FV(n,i,P,0),0), finance_eps);  // general
+  EXPECT_NEAR(P, agent_->PV(n*12,i/12,agent_->FV(n*12,i/12,P,0),0), finance_eps);  // general
+  EXPECT_NEAR(P, agent_->PV(-n,i,agent_->FV(-n,i,P,0),0), finance_eps);  // general
+  
+  // manual testing
+  EXPECT_NEAR(A, agent_->FV(n,0,0,A), finance_eps); // discount rate = 0
+  EXPECT_NEAR(0, agent_->FV(0,i,0,A), finance_eps);  // discount time = 0
+  EXPECT_NEAR(A, agent_->FV(1,i,0,A), finance_eps); // discount time = 1
+  EXPECT_NEAR(A * (std::pow((1 + i), n) - 1) / i, agent_->PV(n,i,0,A), finance_eps); // general
+  EXPECT_NEAR(A * (std::pow((1 + i/12), n*12) - 1) / i/12, agent_->PV(n*12,i/12,0,A), finance_eps); // general
+  EXPECT_NEAR(A * (std::pow((1 + i), -n) - 1) / i, agent_->PV(-n,i,0,A), finance_eps); // general
+
+  EXPECT_NEAR(P + A, agent_->FV(n,0,P,A), finance_eps); // discount rate = 0
+  EXPECT_NEAR(P, agent_->FV(0,i,P,A), finance_eps); // discount time = 0
+  EXPECT_NEAR(P * (1+i) + A, agent_->FV(1,i,P,A), finance_eps); // discount time = 1
+  EXPECT_NEAR(P * std::pow((1+i), n) + A * (std::pow((1 + i), n) - 1),
+              agent_->FV(n,i,P,A), finance_eps); // general
+
+}
+
+TEST_P(AgentTests, PMT) {
+
+  int n = 5;
+  double i = 0.1;
+  double P = 1.5;
+  double F = 3.5;
+  double finance_eps = 1e-4;
+
+  // all testing by "round trip" with already tested functions
+
+  EXPECT_NEAR(P, agent_->PV(n,0,0,agent_->PMT(n,0,P,0)), finance_eps); // discount rate = 0
+  EXPECT_NEAR(P, agent_->PV(0,i,0,agent_->PMT(0,i,P,0)), finance_eps); // discount time = 0
+  EXPECT_NEAR(P, agent_->PV(1,i,0,agent_->PMT(1,i,P,0)), finance_eps); // discount time = 1
+  EXPECT_NEAR(P, agent_->PV(n,i,0,agent_->PMT(n,i,P,0)), finance_eps); // general
+  
+  EXPECT_NEAR(F, agent_->FV(n,0,0,agent_->PMT(n,0,0,F)), finance_eps); // discount rate = 0
+  EXPECT_NEAR(F, agent_->FV(0,i,0,agent_->PMT(0,i,0,F)), finance_eps); // discount time = 0
+  EXPECT_NEAR(F, agent_->FV(1,i,0,agent_->PMT(1,i,0,F)), finance_eps); // discount time = 1
+  EXPECT_NEAR(F, agent_->FV(n,i,0,agent_->PMT(n,i,0,F)), finance_eps); // general
+
+  EXPECT_NEAR(P + agent_->PV(n,i,F,0), agent_->PV(n,i,0,agent_->PMT(n,i,P,F)), finance_eps);
+
+}
+
+
+TEST_P(AgentTests, PVA) {
+
+  std::vector<double> A1({1.0,1.0,1.0,1.0,1.0});
+  std::vector<double> A2({1.0,2.0,3.0,3.5,4.71});
+
+  int n = A1.size();
+  double i = 0.1;
+  double finance_eps = 1e-4;
+
+  EXPECT_NEAR(agent_->PV(n,i,0,A1[0]), agent_->PV(i, A1), finance_eps);
+  double PV_A2 = 0;
+  for (int n2=0; n2 < A2.size(); n2++) {
+    PV_A2 += agent_->PV(n2+1, i, 0, A2[n2]);
```

Comment:
Each payment acts like a separate single future value in this case
```suggestion
    PV_A2 += agent_->PV(n2+1, i, A2[n2], 0);
```